### PR TITLE
Implement support for .git/info/exclude

### LINKED ISSUE #2194: Expose the "repository excludes" file for configuration
It would be helpful if Git Extensions offered improved support for [`$GIT_DIR/info/exclude`](http://git-scm.com/docs/gitignore).
- [ ] Currently the Repository menu contains an option **Edit .gitignore**. It would be helpful if the menu contained an additional option **Edit .git/info/exclude**.
- [ ] The context menu for unstaged files in the Commit dialog includes an option **Add file to .gitignore**. It would be helpful if the menu contained an additional option **Add file to .git/info/exclude (ignore locally)**.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #2194

> :warning: This is totally untested, but I figured I could at least push a branch with a working start for this feature.

### CODE REVIEW

File: GitUI/CommandsDialogs/FormGitIgnore.cs [hunk 1]
```diff
@@ -164,7 +189,7 @@ private void AddDefaultClick(object sender, EventArgs e)
             // workaround to prevent GitIgnoreFileLoaded event handling (it causes wrong _originalGitIgnoreFileContent update)
             // TODO: implement in FileViewer separate events for loading text from file and for setting text directly via ViewText
             _NO_TRANSLATE_GitIgnoreEdit.TextLoaded -= GitIgnoreFileLoaded;
-            _NO_TRANSLATE_GitIgnoreEdit.ViewText(".gitignore",
+            _NO_TRANSLATE_GitIgnoreEdit.ViewText(Path.GetFileName(ExcludeFileRelative),
```
Comment (sharwell) [inline GitUI/CommandsDialogs/FormGitIgnore.cs]: :exclamation: This should just be `ExcludeFileRelative`.

File: GitUI/CommandsDialogs/FormGitIgnore.cs [hunk 2]
```diff
@@ -173,7 +198,7 @@ private void AddDefaultClick(object sender, EventArgs e)
         private void AddPattern_Click(object sender, EventArgs e)
         {
             SaveGitIgnore();
-            UICommands.StartAddToGitIgnoreDialog(this, "*.dll");
+            UICommands.StartAddToGitIgnoreDialog(this, false, "*.dll");
```
Comment (jbialobr) [inline GitUI/CommandsDialogs/FormGitIgnore.cs]: Shouldn't be here `_localExclude` instead of `false`?

Comment (sharwell) [inline GitUI/CommandsDialogs/FormGitIgnore.cs]: Fixed!

**Comment by jbialobr** [comment]
The titles of both forms need to be updated depending on `localExclude`. 

2016-07-15 12_34_46-gitextensions master - git extensions -_ debug _-
What about placing the `Edit .git/info/exlude` directly under the `Edit .gitignore` item?

Other than that works good.

**Comment by sharwell** [comment]
> The titles of both forms need to be updated depending on `localExclude`.

Done! (At least I think done... wasn't sure if this is the right place/way to use `TranslationString`...)

> What about placing the `Edit .git/info/exclude` directly under the `Edit .gitignore` item?

This was my original thought. The only issue I thought of is the other three items are all files which get committed to source control, while the local excludes file is separate. Assuming that some users will be unfamiliar with the local exclude file, is there a way we can avoid making this confusing?

**Comment by jbialobr** [comment]
> is there a way we can avoid making this confusing?

I don't know. Users should use features consciously. When I use a new feature I try to gain a decent piece of knowledge about it to understand how it works. It is harder to know all the features git provides than get the knowledge about a single feature. Placing it closely to .gitignore increases the chances that users will get to know about  the `.git/info/exclude`. From the other hand placing items grouped by location is more friendly to the users that already are familiar with those features. But all in all this is a minor issue.

>Done! (At least I think done... wasn't sure if this is the right place/way to use TranslationString...)

The place is right, but I would replace the file name with the '{0}' placeholder like [here](https://github.com/gitextensions/gitextensions/blob/9c3e6aa30c589a7d3784eeba0b108d5b6b5059aa/GitUI/CommandsDialogs/SettingsDialog/Pages/ChecklistSettingsPage.cs#L19) in all the translation strings.

**Comment by pmiossec** [comment]
>is there a way we can avoid making this confusing?

Perhaps add a label in the top of the form to explain a little...

**Comment by jrfnl** [comment]
Two minor usability remarks:
* The links in the `Edit .git/info/exclude` modal refer to `gitignore` not to the `exclude` file:
* Similarly, the `Save changes?` modal which sometimes pops up, also refers to `.gitignore` not `.git/info/exclude`

screenshot

**Comment by jbialobr** [comment]
Please, create a new issue, that we can see it is open and needs to be resolved.

**Comment by jrfnl** [comment]
@jbialobr Done: https://github.com/gitextensions/gitextensions/issues/3860 :sunglasses:

**Comment by jbialobr** [comment]
Thank you.

**Pull request merged**